contracts/PodsUpkeep.sol

```diff
@@ -0,0 +1,119 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.7.6;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts-upgradeable/math/SafeMathUpgradeable.sol";
+
+import "@pooltogether/pooltogether-generic-registry/contracts/AddressRegistry.sol";
+
+import "./interfaces/IPod.sol";
+import "./interfaces/KeeperCompatibleInterface.sol";
+
+
+///@notice Contract implements Chainlink's Upkeep system interface, automating the upkeep of the Pods contract
+contract PodsUpkeep is KeeperCompatibleInterface, Ownable {
+
+    using SafeMathUpgradeable for uint256;
+    
+    /// @notice Address of the registry of pods contract which require upkeep
+    AddressRegistry public podsRegistry; // or should this live in the PodFloatStrategy repo?
+
+    /// @notice Interval at which pods.batch() will be called
+    uint256 public upkeepBlockInterval;    
+
+    /// @notice Block number when batch was last called successfully
+    uint256 public lastUpkeepBlockNumber;
+
+    /// @notice Minimum pod float fraction
+    uint256 public targetFloatFraction;
+
+    /// @notice Emitted when the upkeep block interval is updated
+    event UpkeepBlockIntervalUpdated(uint upkeepBlockInterval);
+        
+    /// @notice Emitted when the float fraction is updated
+    event TargetFloatFractionUpdated(uint targetFloatFraction);
+
+
+    /// @notice Contract Constructor. No initializer. 
+    constructor(AddressRegistry _podsRegistry, address _owner) Ownable() {
+        
+        podsRegistry = _podsRegistry;
+        
+        transferOwnership(_owner);
+
+        lastUpkeepBlockNumber = block.number;
+
+        // initialize values for block interval and float rate?
```

Comment:
@asselstine

Reply:
It seems to me that we need to track the lastUpkeepBlockNumber *per pod*.  Imagine if two pods were out of sync; such that the first pod is always batched, and the second always gets skipped.

Reply:
And to answer your question I don't think we need to initialize it; it'll just try to batch right away

Reply:
While we dont have to initialize the lastUpkeepBlockNumbers (timestamps) we should probably set an initial global upkeepInterval? Save gas on a tx to `setUpkeepBlockInterval()`?